recipes/ffmpeg/all/conanfile.py

```diff
@@ -533,55 +517,76 @@ def opt_append_disable_if_set(args, what, v):
                 self.settings.os, self.settings.os.version))
-            extra_ldflags.append(tools.apple_deployment_target_flag(
+            extra_ldflags.append(apple_deployment_target_flag(
                 self.settings.os, self.settings.os.version))
-        if self._is_msvc:
-            args.append("--pkg-config={}".format(tools.get_env("PKG_CONFIG")))
+        if is_msvc(self):
+            pkg_config = os.getenv("PKG_CONFIG")
+            args.append(f"--pkg-config={pkg_config}")
             args.append("--toolchain=msvc")
-            if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) <= "12":
+            if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) <= "12":
                 # Visual Studio 2013 (and earlier) doesn't support "inline" keyword for C (only for C++)
-                self._autotools.defines.append("inline=__inline")
+                tc.defines.append("inline=__inline")
         if cross_building(self):
             if self._target_os == "emscripten":
                 args.append("--target-os=none")
             else:
-                args.append("--target-os={}".format(self._target_os))
+                args.append(f"--target-os={self._target_os}")
 
             if is_apple_os(self):
                 if self.options.with_audiotoolbox:
                     args.append("--disable-outdev=audiotoolbox")
 
-                xcrun = tools.XCRun(self.settings)
-                apple_arch = tools.to_apple_arch(str(self.settings.arch))
+                xcrun = XCRun(self)
+                apple_arch = to_apple_arch(self)
                 extra_cflags.extend(
-                    ["-arch {}".format(apple_arch), "-isysroot {}".format(xcrun.sdk_path)])
+                    [f"-arch {apple_arch}", f"-isysroot {xcrun.sdk_path}"])
                 extra_ldflags.extend(
-                    ["-arch {}".format(apple_arch), "-isysroot {}".format(xcrun.sdk_path)])
-
-        args.append("--extra-cflags={}".format(" ".join(extra_cflags)))
-        args.append("--extra-ldflags={}".format(" ".join(extra_ldflags)))
-
-        self._autotools.configure(
-            args=args, configure_dir=self._source_subfolder, build=False, host=False, target=False)
-        return self._autotools
+                    [f"-arch {apple_arch}", f"-isysroot {xcrun.sdk_path}"])
+
+        extra_cflags_string = " ".join(extra_cflags)
+        args.append(f"--extra-cflags={extra_cflags_string}")
+        extra_ldflags_string = " ".join(extra_ldflags)
+        args.append(f"--extra-ldflags={extra_ldflags_string}")
+
+        # FIXME: This is a hack that feels wrong but I don't know how to fix properly. Is this an AutotoolsTolchain
+        #  problem
+        del tc.configure_args[7] # --oldincludedir=${prefix}/include
+        del tc.configure_args[6] # --includedir=${prefix}/include
+        del tc.configure_args[4] # --sbin=${prefix}/bin
```

Comment:
Is there a cleaner way to deal with this issue where ffmpeg doesn't accept this arguments ?